command/agent/consul/service_client.go

```diff
@@ -1476,10 +1476,24 @@ func getAddress(addrMode, portLabel string, networks structs.Networks, driverNet
 		mapping, ok := ports.Get(portLabel)
 		if !ok {
 			ip, port := networks.Port(portLabel)
-			if ip == "" && port <= 0 {
+			if ip != "" && port > 0 {
+				return ip, port, nil
+			}
+
+			// If port isn't a label, try to parse it as a literal port number
+			port, err := strconv.Atoi(portLabel)
+			if err != nil {
+				// Don't include Atoi error message as user likely
+				// never intended it to be a numeric and it creates a
+				// confusing error message
 				return "", 0, fmt.Errorf("invalid port %q: port label not found", portLabel)
 			}
-			return ip, port, nil
+			if port <= 0 {
+				return "", 0, fmt.Errorf("invalid port: %q: port must be >0", portLabel)
+			}
+
+			// A number was given which will use the Consul agent's address and the given port
+			return "", port, nil
```

Comment:
Might want to mention a blank ip is a valid return value from this func in the docstring as at a glance L1479 makes it seem like a blank ip is *not* valid.